app/validators/html_validator.rb

```diff
@@ -3,12 +3,16 @@
 class HtmlValidator < ActiveModel::EachValidator
   def validate_each(record, attribute, value)
     return if value.blank?
-    record.errors.add(attribute, I18n.t('html_validator.invalid_markup')) unless valid_html?(value)
+    errors = html_errors(value)
+    unless errors.empty?
+      record.errors.add(attribute, I18n.t('html_validator.invalid_markup', error: errors.first.to_s))
```

Comment:
The error message from Nokogiri won't be localized, so I think it might be worthless to show it to the end-user.